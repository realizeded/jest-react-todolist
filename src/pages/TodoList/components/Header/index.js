import React from 'react';
class Header extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            inputValue:''
        }
        this.handleInputChange = this.handleInputChange.bind(this);
        this.handleInputKeyUp = this.handleInputKeyUp.bind(this);
    }
    handleInputKeyUp(e) {
        if(e.keyCode===13&&this.state.inputValue) {
            this.props.addListItem(this.state.inputValue);
            this.setState({inputValue:''});
        }
    }
    handleInputChange(e) {
        this.setState(()=>{
            return {
                inputValue:e.target.value
            }
        })
    }
    render() {
        const {inputValue} = this.state;
        const {handleInputChange,handleInputKeyUp} = this;
        return (
                <div className="header">
                    <div className="header-content">
                        <div>
                            TodoList
                        </div>
                        <input type="text" data-test="input" placeholder="Add ToDo" value={inputValue} 
                        onKeyUp={handleInputKeyUp}
                        onChange={handleInputChange}/>
                    </div>
                </div>
            );
    }
}
export default Header;