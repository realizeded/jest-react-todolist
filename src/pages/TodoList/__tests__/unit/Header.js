import React from 'react';
import Enzyme,{shallow,mount}from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../../components/Header';
Enzyme.configure({ adapter: new Adapter() });

//对样式做一个快照 从而判断样式是否更改成功
it("header 样式",()=>{
      const wrapper = shallow(<Header/>);
      expect(wrapper).toMatchSnapshot();
})

it('Header 包含一个input框', () => {
      const wrapper = shallow(<Header/>);
      expect(wrapper.find('[data-test="input"]').length).toBe(1);
});

it('Header input框默认值为空', () => {
      const wrapper = shallow(<Header/>);
      expect(wrapper.find('[data-test="input"]').prop('value')).toEqual('');
});
it('Header input值随输入改变', () => {
      const wrapper = shallow(<Header/>);
      const inputEle = wrapper.find('[data-test="input"]');
      inputEle.simulate('change',{
            target:{
                  value:'input'
            }
      });
      expect(wrapper.state('inputValue')).toEqual('input');
      // expect(inputEle.prop('value')).toEqual('');

});
it('Header input回车时如果input值是空的 则不添加', () => {
      const fn = jest.fn();
      const wrapper = shallow(<Header addListItem={fn}/>);
      const inputEle = wrapper.find('[data-test="input"]');
      wrapper.setState({inputValue:''});
      inputEle.simulate('keyup',{
            keyCode:13
      });
      expect(fn).not.toHaveBeenCalled();

});

it('Header input回车时如果input值不是空的 则添加', () => {
      const fn = jest.fn();
      const wrapper = shallow(<Header addListItem={fn}/>);
      const inputEle = wrapper.find('[data-test="input"]');
      wrapper.setState({inputValue:'input'});

      inputEle.simulate('keyup',{
            keyCode:13
      });
      expect(fn).toHaveBeenCalledWith('input');
      expect(wrapper.state('inputValue')).toEqual('');
});