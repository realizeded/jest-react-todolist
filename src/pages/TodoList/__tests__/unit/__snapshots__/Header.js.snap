// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header组件 样式测试是否有被修改 1`] = `
<div
  className="header"
>
  <div
    className="header-content"
  >
    <div>
      TodoList
    </div>
    <input
      data-test="input"
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="Add ToDo"
      type="text"
      value=""
    />
  </div>
</div>
`;
